src/pages/DirectMessage.tsx

```diff
@@ -0,0 +1,252 @@
+
+  useEffect(() => {
+    const fetchDirectMessage = async () => {
+      const result = await axios.get(
+        `http://localhost:3001/api/messages/${userId1}/${userId2}`
+      );
+      setMessages(result.data.messages as Message[]);
+      dispatch(
+        setSelectedConversation({
+          ...result.data.otherUser,
+          userId: Number(userId2),
+        })
+      );
+    };
+    fetchDirectMessage();
+  }, [dispatch, userId1, userId2]);
+
+  useEffect(() => {
+    // TODO: See if this is the best way to scroll to the bottom, and check edge cases
+    if (messageRef.current) {
+      messageRef.current.scrollTo(0, messageRef.current.scrollHeight);
+    }
+  }, [messages]);
+
+  const onSubmit = async (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    try {
+      const newMessage = (
+        await axios.post("http://localhost:3001/api/messages", {
+          sentUserId: user.userId,
+          receivedUserId: selectedConversation.userId,
+          textContent,
+        })
+      ).data as Message;
+      setTextContent("");
+      setMessages([...messages, newMessage]);
+      dispatch(
+        updateConversation({
+          displayName: selectedConversation.displayName,
+          otherUserId: selectedConversation.userId,
+          textContent: newMessage.textContent,
+          timestamp: newMessage.timestamp,
+          username: selectedConversation.username,
+        })
+      );
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  return (
+    <Stack direction="row" sx={styles.root}>
+      <ConversationList />
+      <Divider flexItem orientation="vertical" />
+      <Box sx={styles.directMessageActivityContainer}>
+        <Box sx={styles.directMessageActivityHeader}>
+          <Box sx={{ alignItems: "center", display: "flex" }}>
+            <Box sx={{ padding: 1 }}>
+              <Avatar />
+            </Box>
+            <Box sx={{ padding: 1 }}>
+              <Typography variant="subtitle1">
+                {selectedConversation.displayName}
+              </Typography>
+              <Typography variant="subtitle2">{`@${selectedConversation.username}`}</Typography>
+            </Box>
+          </Box>
+          <IconButton>
+            <InfoOutlinedIcon />
+          </IconButton>
+        </Box>
+        <Divider flexItem />
+        <Box
+          sx={{
+            display: "flex",
+            flex: 1,
+            flexDirection: "column",
+            overflowY: "hidden",
+          }}
+        >
+          <List
+            component="div"
+            ref={messageRef}
+            sx={{ flex: 1, overflowY: "scroll" }}
+          >
+            {messages.map((o) => (
+              <ListItem component="div" key={o.messageId}>
+                <ListItemText
+                  sx={
+                    o.sentUserId === user.userId
+                      ? styles.sentMessage
+                      : styles.message
+                  }
+                  disableTypography
```

Comment:
I don't understand why we need this `disableTypography`, let me know what I might be missing

Reply:
It disables the default styling MUI has. If you don't put `disableTypography`, the primary and secondary text content will be wrapped with a `Typography` component. You can make it work with their `primaryTypographyProps` and `secondaryTypographyProps`, but I chose to scrap their styling and write my own